cmd/sync-rover-groups/ldap.go

```diff
@@ -81,3 +78,57 @@ func (r *ldapGroupResolver) resolve(name string) (*Group, error) {
 		Members: members.List(),
 	}, nil
 }
+
+func (r *ldapGroupResolver) getGitHubUserKerberosIDMapping() (map[string]string, error) {
+	if r.conn == nil {
+		return nil, fmt.Errorf("ldapGroupResolver's connection is nil")
+	}
+
+	searchReq := ldapv3.NewSearchRequest("ou=users,dc=redhat,dc=com", ldapv3.ScopeWholeSubtree, 0, 0, 0,
+		false, "(rhatSocialURL=GitHub*)", []string{"uid", "rhatSocialURL"}, []ldapv3.Control{})
+
+	result, err := r.conn.Search(searchReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search ldap: %w", err)
+	}
+
+	if len(result.Entries) == 0 {
+		logrus.Warn("found no users setting up github URL")
+		return nil, nil
+	}
+
+	ret := map[string]string{}
+
+	for _, entry := range result.Entries {
+		kerberosID := entry.GetAttributeValue("uid")
+		if kerberosID == "" {
+			//should never happen
+			return nil, fmt.Errorf("empty uid from LDAP server")
+		}
+		var gitHubID string
+		for _, value := range entry.GetAttributeValues("rhatSocialURL") {
+			if parsed := getGitHubID(value); parsed != "" {
+				gitHubID = parsed
+			}
+		}
+		if gitHubID == "" {
+			logrus.WithField("kerberosID", kerberosID).WithField("entry", entry).Warn("failed to parse GitHub ID")
+			continue
+		}
+		ret[gitHubID] = kerberosID
+	}
+
+	return ret, nil
+}
+
+func getGitHubID(value string) string {
```

Comment:
Can you put a comment on this function of what the values of `value` could look like? Its a bit confusing that the result is either the last element or the second to last element. I do see the 2 example from your test case, but having it here as well would make this easier to read.

Reply:
Updated.